packages/gatsby-source-drupal/src/utils.js

```diff
@@ -405,17 +407,20 @@ ${JSON.stringify(nodeToUpdate, null, 4)}
   }
 
   for (const node of nodesToUpdate) {
-    if (node.internal.owner) {
-      delete node.internal.owner
-    }
-    if (node.fields) {
-      delete node.fields
+    if (!nodesUpdated.includes(node.id)) {
```

Comment:
This would mean we wouldn't get the later updates? Like are the updates unique and we'd need to merge them together somehow? Or do the later versions of the node not have any updates?

Reply:
also maybe we could catch the self-referencing perhaps when updating references and just ignore those?

Reply:
I think this will make it so a node can only ever be updated 1 time until the Gatsby process restarts, right? Since Gatsby can be a long running process I think this could cause problems where data stops updating.
In WP I filter the updates array so only the last instance of a node update is used. I think you could reverse the updates array and then filter, keeping track of each node ID in a Set, if it's in the set you filter it out of the array (so only the last update for a node is preferred). At the end you can reverse the array again so the updates are in the right order before you apply them.

Reply:
@TylerBarnes does source-wordpress read/write from a temp in-memory version of the nodes? I'm just curious how this works with sequential updates to the same node. gatsby-source-drupal writes to LMDB after each pass of updates which means the next pass gets the updated node.

@smthomas could you add a test w/ the actual fastbuilds data that's triggering this problem?

Reply:
yeah the fastbuilds data has the full entity data so there's only one request